drivers/tripplite_usb.c

```diff
@@ -340,8 +340,11 @@ int match_by_unitid(usb_dev_handle *argudev, USBDevice_t *arghd, usb_ctrl_charbu
 	NUT_UNUSED_VARIABLE(rdbuf);
 	NUT_UNUSED_VARIABLE(rdlen);
 
-	/* Read ups id from the ups.conf */
-    if (value != NULL) {
+    /* If upsid is not defined in the config, return 1 (null behavior - match any device),
+	 * otherwise read it from the device and match against what was asked in ups.conf */
+    if (value == NULL) {
+        return 1;
+    } else {
```

Comment:
I've updated the comment in code to better describe why we `return 1` - hopefully I got it right? ;)

The original wording did not seem unambiguous to me.